src/main/java/com/vaadin/recipes/nonselectablecomboboxitems/NonSelectableComboBoxItems.java

```diff
@@ -0,0 +1,60 @@
+package com.vaadin.recipes.nonselectablecomboboxitems;
+
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.component.dependency.CssImport;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import com.vaadin.flow.router.Route;
+import com.vaadin.recipes.recipe.Metadata;
+import com.vaadin.recipes.recipe.Recipe;
+import com.vaadin.recipes.recipe.Tag;
+
+@Route("non-selectable-combobox-items")
+@Metadata(
+    howdoI = "Create non-selectable items in ComboBox",
+    description = "Create a ComboBox such that some of its items cannot be selected by user.",
+    sourceFiles = { "recipe/nonselectablecomboboxitems/vaadin-combo-box-item-styles.css" },
+    tags = { Tag.JAVA, Tag.THEME }
+)
+@CssImport(
+    themeFor = "vaadin-combo-box-item",
+    value = "./recipe/nonselectablecomboboxitems/vaadin-combo-box-item-styles.css"
+)
+public class NonSelectableComboBoxItems extends Recipe {
+
+
+    public NonSelectableComboBoxItems() {
+        ComboBox<String> combo = new ComboBox<>();
+        combo.setItems("Heading one", "Option one", "Option two",
+                "Option three", "Heading two", "Option four", "Option five");
+
+        combo.setRenderer(new ComponentRenderer<>(item -> {
+            Span span = new Span(item);
+            if (isHeading(item)) {
+                span.getStyle().set("opacity", "0.5");
+            } else {
+                span.getStyle().set("padding-left", "10px");
+                span.getStyle().set("pointer-events", "auto");
+            }
+            return span;
+        }));
+
+        combo.addValueChangeListener(e -> {
+            if ((e.getValue() != null) && isHeading(e.getValue())) {
```

Comment:
This approach will first fire a "Heading one" selection event and then that will be reverted if you actually select the heading, e.g. using keyboard right now. 

How you should you in your application actually listen for value change with this pattern as I suspect adding another value change listener would mean you get "heading" value change events also

Reply:
> I suspect adding another value change listener would mean you get "heading" value change events also

That's right. Therefore, in any other value change listener, one would have to check that the selected item is not a heading before doing anything with it. 

```
combo.addValueChangeListener(e -> {
            if ((e.getValue() != null) && !isHeading(e.getValue())) {
                //do something here
            }
        });
```